blog/_posts/2023-04-20-sbt-plugins-community-repository.md

```diff
@@ -0,0 +1,106 @@
+[here](https://github.com/playframework/playframework/issues/11675#issuecomment-1499869916),
+[here](https://github.com/sbt/sbt/issues/7204), and
+[here](https://github.com/spray/sbt-revolver/issues/112)).
+
+## Retrospective
+
+To understand what happened, here is some background information. sbt plugins are packaged
+as modules published to Ivy or Maven repositories. Historically, sbt plugins used to be
+published primarily to the sbt community repository hosted on Bintray. However, in 2021,
+JFrog has [sunsetted the Bintray
+service](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/).
+Since then, JFrog has granted the Scala Center a cloud-hosted Artifactory instance for free.
+As it was explained in the [release notes of sbt
+1.5.1](https://eed3si9n.com/bintray-to-jfrog-artifactory-migration-status-and-sbt-1.5.1),
+this Artifactory instance has been used as a read-only repository to host all the sbt
+plugins that were published to the community repository at that time, but the recommendation
+to plugin authors was to switch to the Central Maven repository.
+
+In practice, the Artifactory instance sponsored by JFrog expires automatically after a
+year, and we have to explicitly ask for renewal. What happened the [7th of
+April](https://github.com/sbt/sbt/issues/7202) is that the instance expired without
+prior notice. Thankfully, Eugene Yokota, the maintainer of sbt, promptly asked JFrog to
+renew the Artifactory instance, which they generously did. After a couple more technical
+steps, everything was back to normal.
+
+## Current State
+
+Thanks to the efforts of Eugene Yokota and the support from the company JFrog, the situation
+is now back to normal: the read-only repository `repo.scala-sbt.org` backed by an Artifactory
+instance works again and all the Scala builds should be able to resolve their plugin
+dependencies as before.
+
+However, as mentioned by Seth Tisue in [a
+comment](https://github.com/sbt/sbt/issues/7202#issuecomment-1500220344), the event
+highlighted a weakness in the Scala ecosystem infrastructure, which is well illustrated by
+this comic:
+
+<p style="text-align: center">
+  <img style="max-width: 385px" src="https://imgs.xkcd.com/comics/dependency.png" alt="Dependency" />
+</p>
+
+In practice, what do we rely on the sbt community repository for?
+
+The sbt community repository is primarily used for two purposes:
+
+1. It hosts “old” versions of sbt plugins that may still be used today in Scala projects.
+   It is important to note that those sbt plugins are not necessarily that old. Some of
+   them are stable and they simply didn't get a new release in the past two years,
+   meaning that they had no chances to migrate to the Central Maven repository.
+2. It also hosts the Linux packages (deb and rpm) of sbt releases. Note that that part
+   is not read-only: we still publish Linux packages of new sbt releases there. These
```

Comment:
```suggestion
   is not read-only: we still publish new sbt releases there. These
```

Reply:
But also, do you want to say "we"? It's Eugene that publishes them, not the Scala Center.

There might be other "we"s in the post?